src/sources/socket/mod.rs

```diff
@@ -202,23 +212,118 @@ impl SourceConfig for SocketConfig {
 
                 let decoder = DecodingConfig::new(framing, decoding, LogNamespace::Legacy).build();
 
-                let host_key = config
-                    .host_key
-                    .unwrap_or_else(|| log_schema().host_key().to_string());
-                unix::unix_stream(
-                    config.path,
-                    config.socket_file_mode,
-                    host_key,
-                    decoder,
-                    cx.shutdown,
-                    cx.out,
-                )
+                let log_namespace = cx.log_namespace(config.log_namespace);
+
+                unix::unix_stream(config, decoder, cx.shutdown, cx.out, log_namespace)
             }
         }
     }
 
-    fn outputs(&self, _global_log_namespace: LogNamespace) -> Vec<Output> {
-        vec![Output::default(self.output_type())]
+    fn outputs(&self, global_log_namespace: LogNamespace) -> Vec<Output> {
+        let log_namespace = global_log_namespace.merge(Some(self.log_namespace()));
+
+        let schema_definition = self
+            .decoding()
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata();
+
+        let schema_definition = match &self.mode {
+            Mode::Tcp(config) => {
+                let host_key_path = config
+                    .host_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                let port_key_path = config
+                    .port_key()
+                    .as_ref()
+                    .map(|x| owned_value_path!(x))
+                    .map(LegacyKey::InsertIfEmpty);
+
+                schema_definition
+                    .with_source_metadata(
+                        Self::NAME,
+                        host_key_path,
+                        &owned_value_path!(log_schema().host_key()),
```

Comment:
This shouldn't be user configurable in the vector namespace, and instead should be hardcoded.
```suggestion
                        &owned_value_path!("host"),
```